### PR TITLE
cloud backend test: retry creating secrets manager

### PR DESCRIPTION
**Pull request by tgummerer**
Creating the secret manager seems to flake frequently for reasons seemingly unrelated to the test itself, but rather cloud provider flakyness.  In particular the AWS backend seems to be somewhat flaky.

Try to work around this by retrying a few times with a backoff.

Fixes https://github.com/pulumi/pulumi/issues/16410 (hopefully).

### CODE REVIEW

File: pkg/secrets/cloud/manager_test.go [hunk 1]
```diff
@@ -35,7 +37,15 @@ func testURL(ctx context.Context, t *testing.T, url string) {
 	info := &workspace.ProjectStack{}
 	info.SecretsProvider = url
 
-	manager, err := NewCloudSecretsManager(info, url, false)
+	var err error
+	var manager secrets.Manager
+	for i := 1; i < 10; i++ {
```
Comment (julienp) [inline pkg/secrets/cloud/manager_test.go:42]: Should we have a comment that the loop is there to fight flakes?

**Review by julienp — APPROVED** [review]

File: pkg/secrets/cloud/manager_test.go (hunk 1 above)
Comment (justinvp) [reply pkg/secrets/cloud/manager_test.go:42]: ➕

Comment (justinvp) [reply pkg/secrets/cloud/manager_test.go:42]: Yeah, in general, I'd prefer a comment for these types of things to inform the reader why it's there.

**Comment by justinvp** [comment]
> but rather cloud provider flakyness

Could this be a case where we should consider adding some retries in the product itself?

**Comment by tgummerer** [comment]
Hmm yeah adding the retries in the product might be a good idea :thinking: 

Let me try to rejig the PR a little.

File: pkg/secrets/cloud/manager_test.go (hunk 1 above)
Comment (tgummerer) [reply pkg/secrets/cloud/manager_test.go:42]: Makes sense, I'll add a comment.

**Comment by tgummerer** [comment]
Actually on second thought, I'm not sure we should retry here in the product.  Looking back at the error message we're getting it looks like the account sometimes doesn't have permission for some reason.  Could it be our cleanup script rather than the cloud providers fault? Honestly I don't have enough AWS knowledge to be quite sure what exactly is happening here.

**Review by justinvp — APPROVED** [review]

**Pull request merged**